src/components/AuthForm.vue

```diff
@@ -0,0 +1,209 @@
+<template>
+  <div class="auth">
+    <h1 class="auth__title">{{ props.title }}</h1>
+    <p class="auth__subtitle">{{ props.subtitleText }}</p>
+
+    <form @submit.prevent="submitForm" class="auth__form" novalidate>
+      <v-text-field
+        v-model="formData.email"
+        class="auth__input"
+        label="Email"
+        variant="outlined"
+        :error-messages="formErrors.email"
+      ></v-text-field>
+      <br />
+      <v-text-field
+        v-model="formData.password"
+        class="auth__input"
+        label="Password"
+        variant="outlined"
+        :type="showPassword ? 'text' : 'password'"
+        :append-inner-icon="showPassword ? 'mdi-eye' : 'mdi-eye-off'"
+        @click:append-inner="handleShowPassword"
+        :error-messages="formErrors.password"
+      ></v-text-field>
+      <div class="auth__form-error">
+        <p v-if="serverError" class="auth__form-error-text">
+          {{ serverError }}
+        </p>
+      </div>
+      <v-btn
+        type="submit"
+        class="auth__button"
+        :disabled="isLoading || v$.$errors.length > 0"
+      >
+        {{ props.buttonText }}
+      </v-btn>
+    </form>
+    <router-link :to="handleAuthLink" class="auth__link">{{
+      props.linkText
+    }}</router-link>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { computed, reactive, ref } from "vue";
+import { useRoute, useRouter } from "vue-router";
+import useVuelidate from "@vuelidate/core";
+import { required, minLength, email, helpers } from "@vuelidate/validators";
+import {
+  EMAIL_REQUIRED_FIELD,
+  EMAIL_TYPE_FIELD,
+  PASSWORD_MINLENGTH_FIELD,
+  PASSWORD_REQUIRED_FIELD,
+} from "../constants/errorMessage";
+import { IForm } from "@/types/IForm";
+import { useValidationErrors } from "@/composables/useValidationErrors";
+
+const props = defineProps({
+  title: { type: String, required: true },
+  buttonText: { type: String, required: true },
+  subtitleText: { type: String, required: true },
+  linkText: { type: String, required: true },
+});
+
+const route = useRoute();
+const router = useRouter();
+
+const formData = reactive<IForm>({
+  email: "",
+  password: "",
+});
+const isLoading = ref<boolean>(false);
+const serverError = ref<string | null>(null);
+const showPassword = ref<boolean>(false);
+
+const handleAuthLink = computed<string>(() => {
+  return route.fullPath === "/sign-in" ? "/sign-up" : "/sign-in";
+});
+
+const rules = computed(() => {
```

Comment:
It's better to define two different rules objects (first one - for SignIn with required checks only, second - for SignUp with more rules)